lib/iris/analysis/maths.py

```diff
@@ -658,7 +658,7 @@ def _math_op_common(cube, operation_function, new_unit, in_place=False):
             operation_function(new_cube._my_data, out=new_cube._my_data)
         except TypeError:
```

Comment:
Proposal: remove the try except. The new meaning of "inplace" is "operate on the cube provided" not necessarily "operate on the data provided".

Therefore this should become:

```
new_cube._my_data = operation_function(new_cube._my_data)
```